test/e2e/tests/confirmations/signatures/sign-typed-data-v4.spec.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  DAPP_HOST_ADDRESS,
+  WINDOW_TITLES,
+  openDapp,
+  unlockWallet,
+} from '../../../helpers';
+import { Ganache } from '../../../seeder/ganache';
+import { Driver } from '../../../webdriver/driver';
+
+describe('Confirmation Signature - Sign Typed Data V4', function (this: Suite) {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) { return; }
+
+  it('initiates and confirms', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver, ganacheServer }: { driver: Driver, ganacheServer: Ganache }) => {
+        const addresses = await ganacheServer.getAccounts();
+        const publicAddress = addresses?.[0] as string;
+
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#signTypedDataV4');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await assertInfoValues(driver);
+
+        await driver.clickElement('[data-testid="confirm-footer-button"]');
+
+        /**
+         * TODO: test scroll and fixing scroll
+         * @see {@link https://github.com/MetaMask/MetaMask-planning/issues/2458}
+         */
+        // test "confirm-footer-button" is disabled and unclickable
+        //
+        // await driver.clickElement('.confirm-scroll-to-bottom__button');
+        // await driver.clickElement('[data-testid="confirm-footer-button"]');
+
+        await assertVerifiedResults(driver, publicAddress);
+      },
+    );
+  });
+
+  it('initiates and rejects', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver, ganacheServer }: { driver: Driver, ganacheServer: Ganache }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#signTypedDataV4');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog, { timeout: 20000 });
+        await driver.clickElement('[data-testid="confirm-footer-cancel-button"]');
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const rejectionResult = await driver.waitForSelector({
+          css: '#signTypedDataV4Result',
+          text: 'Error: User rejected the request.',
+        });
+        assert.ok(rejectionResult);
+      },
+    );
+  });
+});
+
+async function assertInfoValues(driver: Driver) {
+  const origin = driver.findElement({ text: DAPP_HOST_ADDRESS });
+  const contractPetName = driver.findElement({
+    css: '.name__value',
+    text: '0xCcCCc...ccccC',
+  });
+
+  const primaryType = driver.findElement({ text: 'Mail' });
+  const contents = driver.findElement({ text: 'Hello, Bob!' });
+  const fromName = driver.findElement({ text: 'Cow' });
+  const fromAddressNum0 = driver.findElement({ css: '.name__value', text: '0xCD2a3...DD826' });
+  const toName = driver.findElement({ text: 'Bob' });
+  const toAddressNum2 = driver.findElement({ css: '.name__value', text: '0xB0B0b...00000' });
```

Comment:
Could this be abstracted to a reusable `findName` across the files?

Reply:
what if we create a resuable const?
```
const selectorPetName = '.name__value'
```

This way, when reading the the line, we understand what it is doing. Not sure if it's worth needing to view `findName` or `findElementByPetName` to understand the logic

Reply:
This is part of the abstractions so not urgent and can be included in the future work.